qiskit/visualization/latex.py

```diff
@@ -508,7 +515,6 @@ def _build_latex_array(self, aliases=None):
         We will deduce the register names and sizes from aliases.
         """
         columns = 1
-        is_occupied = [False] * self.img_width
```

Comment:
Is the reason this being remove (and all the associated logic below) because we rely on the layers to determine when we need to shift because of a barrier? If so this will be useful because tracking this was a real pain and a constant source of bugs.

Reply:
Yes the layers say which column things should be drawn in, so there is no need to keep track of which spaces are occupied anymore.